tuf/api/serialization/json.py

```diff
@@ -47,19 +49,31 @@ class JSONSerializer(MetadataSerializer):
     Attributes:
         compact: A boolean indicating if the JSON bytes generated in
                 'serialize' should be compact by excluding whitespace.
+        validate: Whether to validate when serializing.
 
     """
 
-    def __init__(self, compact: bool = False):
+    def __init__(self, compact: bool = False, validate: Optional[bool] = False):
         self.compact = compact
+        self.validate = validate
 
     def serialize(self, metadata_obj: Metadata) -> bytes:
         """Serialize Metadata object into utf-8 encoded JSON bytes."""
+
+        md_dict = metadata_obj.to_dict()
+        if self.validate:
+            try:
+                new_md_obj = Metadata.from_dict(copy.deepcopy(md_dict))
+                if metadata_obj != new_md_obj:
+                    raise ValueError("Objects are not equal")
```

Comment:
Is it worth to mention what objects are not equal? The caller might not know how we validate here. What about
```suggestion
                    raise ValueError("metadata_obj != deserialize(serialize(metadata_obj))")
```
Is that to technical?  Alternatively (or in addition) we could also explain what `validate` means above in the attribute docstring.

Reply:
+1 for doscstring improvement: it doesn't even necessarily have to be in the `validate` argument docstring but could be in serialize() or JSONSerializer object docstring... wherever it seems to make sense